app/models/product.rb

```diff
@@ -18,7 +18,8 @@ class Product < ActiveRecord::Base
   # Allow us to use `product.hidden?`
   alias_attribute :hidden, :is_hidden
 
-  validates_presence_of :name, :type
+  validates :type, presence: true
+  validates :name, presence: true, length: { maximum: 200 }
```

Comment:
It's what's in the database. Simple_form seems to read that info from the database, because it does add `maxlength=200` to the input. This just adds the validation in case someone messes with the form.